sdk/core/azure-core/src/main/java/com/azure/core/credential/SimpleTokenCache.java

```diff
@@ -126,19 +136,24 @@ public Mono<AccessToken> getToken() {
         });
     }
 
-    private static String refreshLog(AccessToken cache, OffsetDateTime now, String log) {
-        StringBuilder info = new StringBuilder(log);
+    private static String refreshLog(AccessToken cache, OffsetDateTime now, String log, boolean acquired) {
         if (cache == null) {
-            info.append(".");
-        } else {
-            Duration tte = Duration.between(now, cache.getExpiresAt());
-            info.append(" at ").append(tte.abs().getSeconds()).append(" seconds ")
-                .append(tte.isNegative() ? "after" : "before").append(" expiry. ")
-                .append("Retry may be attempted after ").append(REFRESH_DELAY.getSeconds()).append(" seconds.");
-            if (!tte.isNegative()) {
-                info.append(" The token currently cached will be used.");
-            }
+            return acquired ? NO_CACHE_ACQUIRED : NO_CACHE_FAILED;
+        }
+
+        Duration tte = Duration.between(now, cache.getExpiresAt());
+
+        // Use a StringBuilder large enough where it won't need to be resized.
+        StringBuilder info = new StringBuilder(log.length() + 128).append(log);
+
+        info.append(" at ").append(tte.abs().getSeconds())
+            // Reduce the number of appends by using static Strings.
+            .append(tte.isNegative() ? NEGATIVE_TTE : POSITIVE_TTE);
+
+        if (!tte.isNegative()) {
+            info.append(" The token currently cached will be used.");
```

Comment:
We can update POSITIVE_TTE to include this string. So, we don't need to check `!tte.isNegative()`.